Atomic/atomic.py

```diff
@@ -771,6 +858,371 @@ def install(self):
+        self.writeOut("Extracting to %s" % destination)
+
+        rootfs = os.path.join(destination, "rootfs")
+
+        # Under Atomic, get the real deployment location.  It is needed to create the hard links.
+        try:
+            sysroot = OSTree.Sysroot()
+            sysroot.load()
+            osname = sysroot.get_booted_deployment().get_osname()
+            rootfs = os.path.join("/ostree/deploy/", osname, os.path.relpath(rootfs, "/"))
+        except:
+            pass
+
+        os.makedirs(rootfs)
+        revs = []
+
+        rev = repo.resolve_rev(imagebranch, False)[1]
+
+        manifest = self._get_commit_metadata (repo, rev, "docker.manifest")
+        options = OSTree.RepoCheckoutOptions()
+        options.overwrite_mode = OSTree.RepoCheckoutOverwriteMode.UNION_FILES 
+
+        rootfs_fd = None
+        try:
+            rootfs_fd = os.open(rootfs, os.O_DIRECTORY)
+            if manifest is None:
+                repo.checkout_tree_at(options, rootfs_fd, rootfs, rev)
+            else:
+                layers = self._get_layers_from_manifest(manifest)
+                for layer in layers:
+                    rev_layer = repo.resolve_rev("%s%s" % (OSTREE_DOCKERIMG_PREFIX, layer.replace("sha256:", "")), False)[1]
+                    repo.checkout_tree_at(options, rootfs_fd, rootfs, rev_layer)
+        finally:
+            if rootfs_fd:
+                os.close(rootfs_fd)
+
+        exports = os.path.join(destination, "rootfs/exports")
+
+        if not self.args.display:
+            with open(os.path.join(destination, "image"), 'w') as imgfile:
+                imgfile.write("%s\n" % img)
+            sym = "%s/%s" % (OCI_CHECKOUT_PATH, name)
+            if os.path.exists(sym):
+                os.unlink(sym)
+            os.symlink(destination, sym)
+
+        values = {"DESTDIR" : destination, "NAME" : name}
+        if self.args.setvalues is not None:
+            for i in self.args.setvalues:
+                split = i.find("=")
+                if split < 0:
+                    raise ValueError("Invalid value '%s'.  Expected form NAME=VALUE" % i)
+                key, val = i[:split], i[split+1:]
+                values[key] = val
+
+        def _write_template(data, values, outfile):
+            # A dictionary that returns the empty string by default
+            args = defaultdict(str)
+            args.update(values)
+
+            result = Template(data).substitute(args)
+            outfile.write(result)
+
+        for i in ["config.json", "runtime.json"]:
+            src = os.path.join(exports, i + ".template")
+
+            if os.path.exists(src):
+                with open(src, 'r') as infile, open(os.path.join(destination, i), "w") as outfile:
+                    _write_template(infile.read(), values, outfile)
+
+        unitfile = os.path.join(exports, "service.template")
+        unitfileout = os.path.join("/etc/systemd/system", "%s.service" % name)
```

Comment:
We could probably store `/etc/systemd/system` as a constant as it is being reused.